### PR TITLE
Add Custom migration template into the documentation

### PR DESCRIPTION
**Pull request by Holicz**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no
| Fixed issues | #1122

#### Summary

Added documentation for the custom template. Grammar was checked using grammarly.com but please be aware english is not my first language.

### CODE REVIEW

File: docs/en/reference/custom-configuration.rst [hunk 1]
```diff
@@ -127,3 +127,47 @@ With this configuration you can use the ``--conn`` parameter to specify a connec
 migrations. If the parameter is not passed, it will fallback to the one passed in the configuration,
 and if that is also not provided it will fallback to the default connection name specified when creating
 the connection registry.
+
+Custom migration template
+-------------------------
+
+When the default generated migrations do not suit your needs you may provide a custom migration template that will
```
Comment (greg0ire) [inline docs/en/reference/custom-configuration.rst]:
```suggestion
When the default generated migrations do not suit your needs, you may provide a custom migration template that will
```

File: docs/en/reference/custom-configuration.rst [hunk 2]
```diff
@@ -127,3 +127,47 @@ With this configuration you can use the ``--conn`` parameter to specify a connec
...
+be used to generate future migrations.
+
+For example, if you don't need ``down`` migration your template could look like this:
```
Comment (greg0ire) [inline docs/en/reference/custom-configuration.rst]:
```suggestion
For example, if you don't need a ``down`` migration your template could look like this:
```

File: docs/en/reference/custom-configuration.rst [hunk 3]
```diff
@@ -127,3 +127,47 @@ With this configuration you can use the ``--conn`` parameter to specify a connec
...
+
+    final class <className> extends AbstractMigration
+    {
+        public function up(Schema $schema): void
+        {
+            <up>
+        }
+    }
+
+Wildcards (words inside ``< >``) are replaced with correct code upon generation. All possible wildcards are:
```
Comment (greg0ire) [inline docs/en/reference/custom-configuration.rst]:
```suggestion
Placeholders (words inside ``< >``) are replaced with correct code upon generation. All possible wildcards are:
```

File: docs/en/reference/custom-configuration.rst [hunk 4]
```diff
@@ -127,3 +127,47 @@ With this configuration you can use the ``--conn`` parameter to specify a connec
...
+
+===============  ===============================================
+Wildcard         Description
```
Comment (greg0ire) [inline docs/en/reference/custom-configuration.rst]:
```suggestion
Placeholder      Description
```

File: docs/en/reference/custom-configuration.rst [hunk 5]
```diff
@@ -127,3 +127,47 @@ With this configuration you can use the ``--conn`` parameter to specify a connec
...
+---------------  -----------------------------------------------
+``<namespace>``  Namespace of the class, e.g. ``App\Migrations``
+``<className>``  Classname, e.g. `Version20210212092730`
```
Comment (greg0ire) [inline docs/en/reference/custom-configuration.rst]:
```suggestion
``<className>``  Classname, e.g. ``Version20210212092730``
```

**Comment by greg0ire** [comment]
Please associate your email address with your Github account, or change the
email in your commits to an address already associated with it. If you do not
want to expose your personal email address, you may use
`$userName@users.noreply.github.com`, that way we can still reach you through Github.
Additionally, you will get credit for this contribution on your Github profile.

**Comment by Holicz** [comment]
Thank you for your time and the review! I accepted all of the comments and verified my email on Github.

**Review by greg0ire — APPROVED** [review]

**Review by SenseException — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @Holicz !

**Comment by Holicz** [comment]
I see the documentation [was deployed](https://www.doctrine-project.org/projects/doctrine-migrations/en/3.1/reference/custom-configuration.html#custom-configuration), but the table with all available placeholders is not rendered.

I used a simple table according to [this documentation](https://www.doctrine-project.org/projects/doctrine-migrations/en/3.1/reference/custom-configuration.html#custom-configuration), should I rewrite it to the grid table syntax in another PR? I checked some other parts of the documentation and the grid table is rendered correctly.

**Comment by greg0ire** [comment]
Yes, please go ahead.

**Pull request merged**